### PR TITLE
Test pipeline with sklearn and xgboost

### PR DESCRIPTION
**Pull request by apivovarov**
Test DLR model pipeline with sklearn and xgboost models. The models has dynamic batch dim

### CODE REVIEW

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  EXPECT_EQ(shape0[0], 3);
+  EXPECT_EQ(shape0[1], 1);
+  const char* output_type0;
+  EXPECT_EQ(GetDLROutputType(&model, 0, &output_type0), 0);
+  EXPECT_STREQ(output_type0, "float32");
+  // output 0 data
+  float output0[3];
+  EXPECT_EQ(GetDLROutput(&model, 0, output0), 0);
+  // batch 0, 1 and 2
+  EXPECT_LE(output0[0] - 0.7462196, 1e-6);
```
Comment (trevor-m) [inline tests/cpp/dlr_pipeline_skl_xgb_test.cc:152]: Let's do `EXPECT_LE(std::abs(output0[0] - 0.926896), 1e-6);

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestSetDLRInput) {
+  auto model = GetDLRModel();
+  std::string in_data = "[[0.5,0.6],[0.55,0.66],[0.73,0.83]]";
+  int64_t shape[1] = { (int64_t) in_data.length() };
+  int ndim = 1;
+  const char* input_name = "input";
+  EXPECT_EQ(SetDLRInput(&model, input_name, shape, (void*) in_data.c_str(), ndim), 0);
```
Comment (trevor-m) [inline tests/cpp/dlr_pipeline_skl_xgb_test.cc:57]: Let's use C++ style casts. This one should be `const_cast<char*>(in_data.data())`

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  const char* input_type;
+  EXPECT_EQ(GetDLRInputType(&model, 0, &input_type), 0);
+  EXPECT_STREQ(input_type, "json");
+  DeleteDLRModel(&model);
+}
+
+TEST(PipelineTest, TestSetDLRInput) {
+  auto model = GetDLRModel();
+  std::string in_data = "[[0.5,0.6],[0.55,0.66],[0.73,0.83]]";
+  int64_t shape[1] = { (int64_t) in_data.length() };
```
Comment (trevor-m) [inline tests/cpp/dlr_pipeline_skl_xgb_test.cc:54]: Use C++ casts here and elsewhere `static_cast<int64_t>(in_data.length())`

**Review by trevor-m — CHANGES_REQUESTED** [review]
Thanks @apivovarov! Some small changes.

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc (hunk 1 above)
Comment (apivovarov) [reply tests/cpp/dlr_pipeline_skl_xgb_test.cc]: Better to use `EXPECT_FLOAT_EQ`

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc (hunk 2 above)
Comment (apivovarov) [reply tests/cpp/dlr_pipeline_skl_xgb_test.cc]: fixed

File: tests/cpp/dlr_pipeline_skl_xgb_test.cc (hunk 3 above)
Comment (apivovarov) [reply tests/cpp/dlr_pipeline_skl_xgb_test.cc]: fixed

**Review by trevor-m — APPROVED** [review]
Thanks!
